### ISSUE TITLE
Feature request: Skip assets

### ISSUE THREAD

**Issue by cb-lixar** [iBotPeaches/Apktool#1605]
tl;dr: It would be nice to be able to add e.g. -a to the decode command to skip assets altogether.

### Information
1. **Apktool Version (`apktool -version`)** - last checked 2.2.2
2. **Operating System (Mac, Linux, Windows)** - Windows
3. **APK From? (Playstore, ROM, Other)** - Any

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? Yes
2. If you are trying to install a modified apk, did you resign it? N/A
3. Are you using the latest apktool version? No, but changelog doesn't indicate feature has been added.

Short and simple -- would like to be able to skip extracting assets. As far as I've seen personally these can usually be extracted with a simple unzip tool, but regardless, extracting the assets is (at least in situations where I've been using apktool) often unnecessary and in some cases flat out undesirable (for instance, I have an APK here (which I don't think I can share) that's 260MB compressed, 1.6GB uncompressed, and all we're interested in in this particular case is some of the metadata).

**Comment by iBotPeaches** [comment]
Currently, we have 2 options that either skip sources or resources. 

As you probably know, if we skip resources than we just pass the `resources.arsc` file onward to the decode. It then is copied back during compile.

Likewise for the sources, if enabled we just pass the `classes#.dex(s)` onward instead of disassembling them. 

So enter assets. Right now we move them to the filesystem while remembering their compression. If we skipped assets they would be stuck in the original application and to the best of my knowledge short of making an "asset" zip in root, there is no way around keeping the compressed files outside the zip.

I'm not sure for the best course of action yet for this feature request.

**Comment by cb-lixar** [comment]
(I responded to this earlier but either I can't reply-to-comment like Github seems to imply, something went wrong, or it went to your personal email instead or something.)

What I would expect in this case is that they are simply not extracted. Yes, this would mean that you can't rebuild the app as-is with the directory where everything else was extracted, but I'm not sure how else the feature would be interpreted. If it's a question of terminology, "don't extract" could be used instead of "skip". I doubt that in every usage of apktool the intent is to later rebuild the app.

If there were really intent to rebuild, I dunno, I'd say there'd need to be a separate feature for that that would request the original APK or something. At surface value I'd say rebuilding & skipping assets are simply mutually exclusive.

**Comment by iBotPeaches** [comment]
We have `-m / --match-original` which is used for analysis. It prevents any of the "fixes" apktool does to allow rebuild. However, analysis sometimes is required for the unknown/asset files so it doesn't make sense to add this feature to that option (Just something I was thinking)

I don't have a problem adding an additional parameter for this. It may be an advanced command (thus only long version like `--skip-assets` or something). I don't want to pollute the shorthand command parameters anymore unless crucial and since this command won't act the same as the skip resources/sources it makes sense to separate it.

Let me get https://github.com/iBotPeaches/Apktool/pull/1602 merged first as it touches the CLI/parameters and I can swing back to this.

**Comment by cb-lixar** [comment]
Well, it's possible for there to be incompatible arguments; If you want to prevent any confusion then `-m / --match-original` could block/warn usage of `--skip-assets` (or whatever the name ends up being) with some error/warning message explaining the conflict.

At any rate, that all sounds good to me and `--skip-assets` or `--no-assets` would make sense to me. I appreciate the quick responses on this, too.

**Comment by cb-lixar** [comment]
So I can get an idea of if I should look for alternatives in the interim, when do you think you would be able to work on this? (I'm not trying to add any pressure.)

**Comment by iBotPeaches** [comment]
I have Thursday set aside this week at work to work on Apktool. I will fix up the pending PRs and see if I can add this real quick. That would seal it into the 2.3 release coming hopefully early next week.

**Comment by cb-lixar** [comment]
Any luck on this?

**Comment by iBotPeaches** [comment]
Done. Commit above. It uses `--no-assets` and prevents copying of the `assets` folder.

**Comment by cb-lixar** [comment]
Thanks! I noticed you have a CI but I don't see any way to download the artefact / build from it. Is it possible?

**Comment by iBotPeaches** [comment]
Unfortunately not, there is no publishing attached to our CI. I am planning a release for Wed/Thur of this week.
